### PR TITLE
Cache Google subnets via cats.

### PR DESCRIPTION
**Pull request by duftler**
Add cloudProvider attribute to Subnet interface.
Modify SubnetController list() entrypoint to return a map of cloudProvider -> list-of-subnets.
Add listByCloudProvider() entrypoint to SubnetController.
Update AmazonSubnetProvider to only retrieve 'aws' identifiers.
Update GoogleNetworkProvider to only retrieve 'gce' identifiers.

### CODE REVIEW

**Comment by duftler** [comment]
@ajordens Please review. Have tested this behind deck and verified no breaking changes.

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy [hunk 1]
```diff
@@ -31,10 +32,27 @@ class SubnetController {
   List<SubnetProvider> subnetProviders
 
   @RequestMapping(method = RequestMethod.GET)
-  Set<Subnet> list() {
-    subnetProviders.collectMany {
+  Map<String, Set<Subnet>> list() {
```
Comment (ajordens) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy]: Feels like we need to preserve the original `Set<Subnet>` response and avoid a breaking change. The `Subnet` should have a `type` attribute allowing a prospective caller to differentiate by type if necessary.

Similarly, it seems like filtering by `cloudProvider` could also be accomplished by a filter query parameter? That said, I don't believe we're terribly consistent on filters vs. path params so if you have a reasonable "it's like this already" argument for `/{cloudProvider}` than I'm convinced.

Gate is deserving of a `v2` treatment with consistency in mind.

Comment (duftler) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy]: I found this style (i.e. /{cloudProvider}) in use by CacheController, ClusterController, NetworkController and OperationsController.
The cloudProvider attribute can be used for filtering if necessary by the caller, but would be a breaking change to rely on the caller to do that anyway.
Chose to explicitly break (and fix) gate as the only caller of clouddriver, and not break any callers of gate.
This overall pattern matches that of NetworkController.

Comment (duftler) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy]: Just spotted your comment on the companion PR. Let me have another look at it, and will ping this thread again shortly.

**Comment by duftler** [comment]
Changed to preserve the original behavior of `list()`.
This entry-point isn't used by gate after the pending gate pr is merged, so nothing needs to change on the calling side.
This pr needs to be deployed first.

**Comment by duftler** [comment]
It preserves the original behavior of `list()`, but each subnet contains the `cloudProvider` attribute now.

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy (hunk 1 above)
Comment (ajordens) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SubnetController.groovy]: Argument makes sense to me.

**Comment by ajordens** [comment]
LGTM with this being a non-breaking change now.

**Comment by ajordens** [comment]
With regards to the `cloudProvider` attribute, it's `type` for Cluster/ServerGroup/LoadBalancer.

I'd suggest going with `type` vs. `cloudProvider` for consistency.

**Comment by ajordens** [comment]
LGTM otherwise.

**Comment by duftler** [comment]
Replaced `cloudDriver` with `type`.
@JargoonPard fyi Had to update `AzureSubnet` and `AzureSubnetProvider` as well to satisfy the interfaces.

**Comment by JargoonPard** [comment]
Thanks @duftler

**Pull request merged**